docs/source/concepts/InstrumentAccessLayers.rst

```diff
@@ -117,6 +120,11 @@ ___________
 * Get the ``SpectrumInfo`` object as a const reference and use auto - ``const auto &spectrumInfo = ws->spectrumInfo();``.
 * Do not forget to add the import - ``#include "MantidAPI/SpectrumInfo.h"``.
 
+Performance Tests
+_________________
+
+Please add performance test for any algorithms that are widely used, or might be expected to have a performance increase. See `this performance test <https://github.com/mantidproject/mantid/pull/18189/files#diff-5695221d30495359738f90b83ceb0ba3>`_ added for the ``SpectrumInfo`` rollout for an example of adding such a test.
```

Comment:
I have pushed some changes that hopefully emphasise the importance of this a bit more. I have left in the bit about widely used algorithms. We should just make sure when reviewing the PRs they have been added when sensible.